aperturedb/Connector.py

```diff
@@ -167,11 +167,7 @@ def authenticate(self, shared_data, user, password, token):
         if not self.authenticated:
             if shared_data.session is None:
                 self.shared_data.lock = Lock()
-                try:
-                    self._authenticate(user, password, token)
-                except Exception as e:
-                    raise UnauthenticatedException(
-                        "Authentication failed:", str(e))
+                self._authenticate(user, password, token)
```

Comment:
This means that authenticate will just return on a failed authenticate?

Reply:
All the retry, with connect and refresh had been moved inside common function _query, which takes care of the retry mechanism. This exception need not be handled anywhere, and the one raised in _query will be the one to catch and validate.